### PR TITLE
Refactored hook handlers in mod_last

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR changes all hook handlers in `mod_last` module to `gen_hook` format.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/151060) / small_tests / 4eb12cfc5e4e3c95efb17b0147ae102fc28be20b
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3853/151060/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3853/151060/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-10_10.21.49/lib.mongooseim.logs/run.2022-11-10_10.21.50/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/151082) / small_tests / 2b15d87c9e051b8fb7075ce58afbfa2cfa03c90a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3853/151082/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3853/151082/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-10_11.36.23/lib.mongooseim.logs/run.2022-11-10_11.36.24/suite.log.html)

File: src/mod_last.erl [hunk 1]
```diff
@@ -55,8 +55,7 @@
 -export([config_metrics/1]).
 
 -ignore_xref([
-    behaviour_info/1, on_presence_update/5, process_local_iq/4,
-    process_sm_iq/4, remove_user/3, session_cleanup/5, remove_domain/3
+    behaviour_info/1, process_local_iq/4, process_sm_iq/4
```
Comment (kamilwaz) [inline src/mod_last.erl:58]: This line can be deleted as: 
* `behaviour_info/1` don't need to be ignored,
* `process_local_iq/4` and `process_sm_iq/4` don't exist.

**Review by kamilwaz — CHANGES_REQUESTED** [review]
Looks good 👍 , just one small comment.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/151138) / small_tests / cda7dbd695477bb7344484d9ddd36f66fc3e7f49
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3853/151138/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3853/151138/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-10_13.10.20/lib.mongooseim.logs/run.2022-11-10_13.10.20/suite.log.html)

File: src/auth/ejabberd_auth_anonymous.erl [hunk 2]
```diff
@@ -180,7 +180,7 @@ purge_hook(true, HostType, LUser, LServer) ->
     Acc :: mongoose_acc:t(),
     Params :: map(),
     Extra :: map().
-session_cleanup(Acc, #{sid := SID, user := LUser, server := LServer}, _Extra) ->
+session_cleanup(Acc, #{sid := SID, jid := #jid{luser = LUser,lserver = LServer}}, _Extra) ->
```
Comment (kamilwaz) [inline src/auth/ejabberd_auth_anonymous.erl:183]:
```suggestion
session_cleanup(Acc, #{sid := SID, jid := #jid{luser = LUser, lserver = LServer}}, _Extra) ->
```

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/153022) / small_tests / 59c7fa2d67683ee5a5f0dd05c66f8e8233193e30
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3853/153022/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3853/153022/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-21_07.42.14/lib.mongooseim.logs/run.2022-11-21_07.42.14/suite.log.html)

**Review by NelsonVides — APPROVED** [review]

**Review by kamilwaz — APPROVED** [review]

**Pull request merged**